peer_manager.go

```diff
@@ -1,173 +1,81 @@
 package ipfscluster
 
 import (
-	"sync"
-	"time"
-
+	"fmt"
+	host "github.com/libp2p/go-libp2p-host"
 	peer "github.com/libp2p/go-libp2p-peer"
 	peerstore "github.com/libp2p/go-libp2p-peerstore"
 	ma "github.com/multiformats/go-multiaddr"
 )
 
-// peerManager is our own local peerstore
+// peerManager provides wrappers peerset control
 type peerManager struct {
-	cluster *Cluster
-	ps      peerstore.Peerstore
-	self    peer.ID
-
-	peermap map[peer.ID]ma.Multiaddr
-	m       sync.RWMutex
+	host host.Host
 }
 
-func newPeerManager(c *Cluster) *peerManager {
-	pm := &peerManager{
-		cluster: c,
-		ps:      c.host.Peerstore(),
-		self:    c.host.ID(),
-	}
-	pm.resetPeers()
-	return pm
+func newPeerManager(h host.Host) *peerManager {
+	return &peerManager{h}
 }
 
-func (pm *peerManager) addPeer(addr ma.Multiaddr, save bool) error {
-	logger.Debugf("adding peer %s", addr)
+func (pm *peerManager) addPeer(addr ma.Multiaddr) error {
+	logger.Debugf("adding peer address %s", addr)
 	pid, decapAddr, err := multiaddrSplit(addr)
 	if err != nil {
 		return err
 	}
-	pm.ps.AddAddr(pid, decapAddr, peerstore.PermanentAddrTTL)
-
-	// Only log these when we are not starting cluster (rpcClient == nil)
-	// They pollute the start up logs redundantly.
-	if !pm.isPeer(pid) && pm.cluster.rpcClient != nil {
-		logger.Infof("new peer: %s", addr.String())
-	}
-
-	pm.m.Lock()
-	pm.peermap[pid] = addr
-	pm.m.Unlock()
-
-	if save {
-		pm.savePeers()
-	}
-
-	logger.Debugf("peers after adding %s", pm.peersAddrs())
+	pm.host.Peerstore().AddAddr(pid, decapAddr, peerstore.PermanentAddrTTL)
 	return nil
 }
 
-func (pm *peerManager) rmPeer(pid peer.ID, save bool) error {
-	logger.Debugf("removing peer %s", pid.Pretty())
-
-	// Seeing our own departure during bootstrap. Ignore that.
-	if pid == pm.self && !pm.cluster.readyB {
-		return nil
-	}
-
-	// remove ourselves, unless:
-	// - we are not ready yet (means we are boostrapping)
-	// - we have been removed (means Shutdown() with LeaveOnShutdown flag)
-	if pid == pm.self && pm.cluster.readyB && !pm.cluster.removed {
```

Comment:
These conditions were added because, during bootstrapping to a cluster to which the node already belonged, our log would be replayed to it and there was the chance that the peer saw it's own departure entry. Thus it would shut itself down during bootstrap. Now we don't log or notify departures ourselves so that's not going to happen (at least here). I don't think this function is getting called anymore, it was just managing our peerset which we don't have anymore.